server/src/services/appVersion.js

```diff
@@ -0,0 +1,70 @@
+const Schmervice = require('@hapipal/schmervice')
+const AppVersionModel = require('../models/v2/AppVersion')
+const { executeQuery } = require('../query/executeQuery')
+
+const getAppVersionQuery = knex =>
+    knex('app_version')
+        .innerJoin(
+            knex.ref('app_version_localised').as('avl'),
+            'avl.app_version_id',
+            'app_version.id'
+        )
+        .innerJoin(
+            knex.ref('app_channel').as('ac'),
+            'ac.app_version_id',
+            'app_version.id'
+        )
+        .innerJoin(knex.ref('channel'), 'ac.channel_id', 'channel.id')
+        .select(
+            'app_version.id',
+            'app_version.version',
+            knex.ref('app_version.app_id').as('appId'),
+            knex.ref('app_version.created_at').as('createdAt'),
+            knex.ref('app_version.source_url'),
+            knex.ref('app_version.demo_url'),
+            knex.ref('channel.name').as('channel'),
+            knex.ref('ac.min_dhis2_version'),
+            knex.ref('ac.max_dhis2_version')
+        )
+
+class AppVersionService extends Schmervice.Service {
+    constructor(server, schmerviceOptions) {
+        super(server, schmerviceOptions)
+    }
+
+    async findByAppId(appId, { filters, pager }, knex) {
+        const query = getAppVersionQuery(knex).where(
+            'app_version.app_id',
+            appId
+        )
+
+        if (filters.getFilter('channel')) {
+            filters.applyOneToQuery(query, 'channel', {
+                overrideColumnName: 'channel.name',
+            })
+        }
+
+        const res = await executeQuery(query, {
+            filters,
+            pager,
+            model: AppVersionModel,
+        })
+
+        console.log('rez', res)
+        return res
+    }
+}
+
+const createAppVersionService = (server, schmerviceOptions) => {
```

Comment:
We could've just done `server.register(AppVersionService)` class directly. But if we want to do some initialisation or add caching down the line, this approach of a "service factory" is more flexible.